bakery/src/tasks/upload-book.js

```diff
@@ -14,6 +14,10 @@ const task = (taskArgs) => {
     throw Error('distBucketPath must represent some directory-like path in s3')
   }
   const distBucketPrefix = `${distBucketPath}${codeVersion}`
+  const cloudfrontUrlWithPrefix = `${cloudfrontUrl}/${distBucketPrefix}`
```

Comment:
Do we care that `cloudfrontUrl` may be `undefined` in the distribution pipeline? (I don't think it functionally impacts anything since the output from this task never gets used there)

Reply:
I had the same thought and had to double check that it wouldn't fail (it seems to resolve to the string 'undefined'). Clearly, in the webhosting pipelines, this value is not necessary, since there is no service to report to. It seems like a code smell that the parameter has no meaning and seems to luckily not break stuff in a webhosting pipeline. Maybe we can figure out a better way to separate the upload-book task behavior between distribution-preview and webhosting pipelines in the near future, since it's probably the hardest task to reason about right now because of that.

Maybe there's a quick refactoring I can do that could help?

Reply:
Yeah -- good point that perhaps this is an indication some form of refactoring is due. It can be in this PR (if you see some low hanging fruit), or we can create a maintenance issue to come back and look at that.

Reply:
Did a bit of shifting around. Overall a bit happier with the upload-book task and its params now.